pkg/kubelet/lifecycle/handlers.go

```diff
@@ -104,6 +105,30 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 	return -1, fmt.Errorf("couldn't find port: %v in %v", portReference, container)
 }
 
+// formatURL formats a URL from args.
+func formatURL(scheme string, host string, port int, path string) *url.URL {
+	u, err := url.Parse(path)
+	// Something is busted with the path, but it's too late to reject it. Pass it along as is.
+	if err != nil {
+		u = &url.URL{
+			Path: path,
+		}
+	}
+	u.Scheme = scheme
+	u.Host = net.JoinHostPort(host, strconv.Itoa(port))
+	return u
+}
+
+// buildHeader takes a list of HTTPHeader <name, value> string pairs
+// and returns a populated string->[]string http.Header map.
+func buildHeader(headerList []v1.HTTPHeader) http.Header {
```

Comment:
Can you share more about the divergence between methods for building headers between your impl and https://github.com/kubernetes/kubernetes/pull/68848?

Reply:
The main goal is to get lifecycle handlers as close to what probers are doing today. With that said, probers use an identical [`buildHeader`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/prober.go#L151-L157) func, so this PR borrows that logic instead of doing what #68848 did.